### PR TITLE
Redis mask dict preparation

### PR DESCRIPTION
**Pull request by Robert-Steiner**
**Summary**
- prepares the Redis client for storing mask dict 
- added lua script for increment mask count
- added PET errors for the increment mask count operation
- fixed/added tests

### CODE REVIEW

**Comment by janpetschexain** [comment]
there seems to be some overlappings with #537 , should we wait until that is merged and rebased before we review this one?

File: rust/xaynet-server/src/storage/redis.rs
Comment (little-dude) [inline rust/xaynet-server/src/storage/redis.rs:363]: seems like this comment is truncated.

File: rust/xaynet-server/src/storage/impls.rs
Comment (little-dude) [inline rust/xaynet-server/src/storage/impls.rs:8]: Looks like a rebasing issue?

**Review by little-dude — APPROVED** [review]
60491ba looks good to me :100: 
As @janpetschexain said the timeout changes have been pulled in though. Maybe we could just cherry-pick 60491ba while we take a decision on those?

File: rust/xaynet-server/src/storage/redis.rs
Comment (Robert-Steiner) [reply rust/xaynet-server/src/storage/redis.rs:363]: Oh yes, good catch! I will fix it

File: rust/xaynet-server/src/storage/impls.rs
Comment (Robert-Steiner) [reply rust/xaynet-server/src/storage/impls.rs:8]: rebasing issue because it was previously `MaskMany` or because is now `MaskObject`?

File: rust/xaynet-server/src/storage/impls.rs
Comment (finiteprods) [inline rust/xaynet-server/src/storage/impls.rs:352]: thought that occurred to me: it would be nice to be able to map "error codes" to error variants right at the point of definition, by making the error codes discriminants of the enum variants i.e.
```rust
pub enum MaskDictIncrError {
    UnknownSumPk = -1,
    MaskAlreadySubmitted = -2,
}
```
but it seems it's not straightforward to convert from an int to an enum, there's at least [some boilerplate](https://stackoverflow.com/a/57578431) required. so I'm not sure whether it's worth it. but might be worth considering.

File: rust/xaynet-server/src/storage/redis.rs
Comment (finiteprods) [inline rust/xaynet-server/src/storage/redis.rs:321]: do you mean "sum participant"?
while we're here, also check the comment on line 249:
```
-- This condition and should never apply.
```

File: rust/xaynet-server/src/storage/redis.rs
Comment (finiteprods) [inline rust/xaynet-server/src/storage/redis.rs:341]: why *two* masks?

**Review by finiteprods — COMMENTED** [review]
looks good. just a few minor comments.

File: rust/xaynet-server/src/storage/impls.rs
Comment (Robert-Steiner) [reply rust/xaynet-server/src/storage/impls.rs:352]: Not a bad idea🤔 Maybe we can use `#[derive(TryInto)]` to hide the boilerplate code. I will try it out and give an update on the result.

File: rust/xaynet-server/src/storage/redis.rs
Comment (Robert-Steiner) [reply rust/xaynet-server/src/storage/redis.rs:341]: In the state unmask, we check whether the two masks with the highest score do not have the same score.

File: rust/xaynet-server/src/storage/redis.rs
Comment (Robert-Steiner) [reply rust/xaynet-server/src/storage/redis.rs]: fixed

**Pull request merged**
